### ISSUE TITLE
Bug v2-2 beta0?

### ISSUE THREAD

**Issue by JeromeLeCoz** [BaRatin-tools/BaRatinAGE#10]
Hello

I think I met a bug in modifying this bar.zip file (created with v2-1). I created a second hydraulic config, then computed the prior RC (OK), then failed to compute the posterior RC ("execution problem"). I can still compute the RC with the first config, but it corresponds to the second config! I fear something is mixed up in the files...

[Ramon_Baratinage.bar.zip](https://github.com/BaRatin-tools/BaRatinAGE/files/10034272/Ramon_Baratinage.bar.zip)

**Comment by JeromeLeCoz** [comment]
I rebuilt the whole study from scratch using v2-2 beta0 and everything works well. I did not use the Duplicate button for configs and RCs as I did previously, so the bug might come from Duplicating the config...

[Ramon.bar.zip](https://github.com/BaRatin-tools/BaRatinAGE/files/10034348/Ramon.bar.zip)

**Comment by JeromeLeCoz** [comment]
Follow-up:

I've just tried to use Duplicate in my v2-2 study and it seems to work properly. The pb may be some incompatibility between v2-1 and v2-2 files.

I've played with the RC extra plots and they all look fine. The new extra plot "Remarks" is very nice. However, in my Ramon case, the message says possible prior/post inconsistency for parameter a2, but when I look at the prior/post distributions, they don't look to be in conflict... Is the criterion too strict?

Minor detail: in the prior/post table, it would be great to have plus/minus signs instead of +/- (if easy to implement).

**Comment by IvanHeriver** [comment]
> Hello
> 
> I think I met a bug in modifying this bar.zip file (created with v2-1). I created a second hydraulic config, then computed the prior RC (OK), then failed to compute the posterior RC ("execution problem"). I can still compute the RC with the first config, but it corresponds to the second config! I fear something is mixed up in the files...
> 
> [Ramon_Baratinage.bar.zip](https://github.com/BaRatin-tools/BaRatinAGE/files/10034272/Ramon_Baratinage.bar.zip)

Thanks for the feedback. I cannot open this project file without error in BaRatinAGE v2.2. It turns out that for the second hydraulic configuration the files `0_control.txt` and `1_control.txt` in `HydraulicConfiguration/1/`which are supposed to contains the prior information for the parameters are not formatted correctly... 

Question: what do you mean by "modyfing this bar.zip file"? Did you create the second control within BaRatinAGE v2.2? If so, do you have the bar.zip file prior to this modification so I can reproduce the steps which led to this bug. thanks

**Comment by IvanHeriver** [comment]
> I rebuilt the whole study from scratch using v2-2 beta0 and everything works well. I did not use the Duplicate button for configs and RCs as I did previously, so the bug might come from Duplicating the config...
> 
> [Ramon.bar.zip](https://github.com/BaRatin-tools/BaRatinAGE/files/10034348/Ramon.bar.zip)

I tried the duplicate feature (i.e. right click on a hydraulic configuration  + duplicate) and everything worked fine for me (prior and posterior rating curve using the duplicated hydraulic config)...

Doesn't seem that the duplicate feature is broken...

**Comment by IvanHeriver** [comment]
> I've just tried to use Duplicate in my v2-2 study and it seems to work properly. The pb may be some incompatibility between v2-1 and v2-2 files.

Difficult to say for sure. Would be great to be able to reproduce the steps that lead to this bug.

**Comment by JeromeLeCoz** [comment]
Yes, I have created the second control. Sorry, I only kept the modified file...

But this test seems to reproduce the bug: create a study with 2-1 with a hydraulic config with a single channel control. Then open it with 2-2, duplicate the hyd config, and change it to 2 channel controls (additive). Then, the prior RC is not computed... Here are the 2-1 and 2-2 files.

[test.bar.zip](https://github.com/BaRatin-tools/BaRatinAGE/files/10121394/test.bar.zip)
[test2-2.bar.zip](https://github.com/BaRatin-tools/BaRatinAGE/files/10121395/test2-2.bar.zip)

**Comment by IvanHeriver** [comment]
Thank you Jérôme, I was able to reproduce the bug. It is not related to the version 2.1 as it occurs also with version 2.2. 

I suspect the bug to be related to duplicating an empty hydraulic configuration (i.e. nothing specified for the parameters).

I think this isn't trivial to fix and more investigation is required to know for sure the source of the problem... Is it worth fixing for v2.2?

**Comment by JeromeLeCoz** [comment]
I think it would be enough in 2-2 to prevent the user from doing what does not work (perhaps with a simple warning explaining that duplicating an empty hydraulic configuration is not allowed?).

**Comment by IvanHeriver** [comment]
I think I corrected the bug. It actually affected much more than duplication of hydraulic configurations.

A more detailed description of the bug would be:
- Any file saved with an empty hydraulic config is correctly saved
- It can then be open with no error however, it is not correctly open (although nothing is obvious in the UI)
- when saved again (save as, save, etc...) and open back, the error will occur.

In other words, duplicating the hydraulic configuration had not much to do with the problem.

Hopefully, the modifications I made fixed the bug.

**Comment by JeromeLeCoz** [comment]
It seems the bug is still there. I repeated the same actions (using 2-1-1 and 2-2) and now get an error message when trying to compute the prior RC:
![image](https://user-images.githubusercontent.com/86300732/205064589-cd840268-f615-4269-904e-3a9d24e1525b.png)

Here's the file:
[test.bar.zip](https://github.com/BaRatin-tools/BaRatinAGE/files/10132717/test.bar.zip)

**Comment by JeromeLeCoz** [comment]
Je précise d'ailleurs que les contrôles ne sont pas vides mais spécifiés, avec CT prior calculées.

**Comment by IvanHeriver** [comment]
I was not able to reproduced the bug.

I tried:
- Open v2.1, create a hydraulic config, keeping it with a single control, use the same parameter values as you did in your test project file, save the file.
- Open v2.2, open the project, duplicate the hydraulic config, configure it as you did with two controls and the same parameter values
- then, after setting up the stage range for the rating curve, clicking on "execute" produces the prior rating curve as expected...

What did I miss?

**Comment by JeromeLeCoz** [comment]
I think my problem is different (and worse): using 2-2 beta1, I cannot run prior RC if I define juste 1 control (channel or weir)...

**Comment by JeromeLeCoz** [comment]
I confirm, and I don't have this problem with 2-2 beta0...

**Comment by IvanHeriver** [comment]
Sorry, forgot to include the BaRatin exe in the beta version I sent you... Should be fixed now. And this should be unrelated to the  issue discussed here.

**Comment by JeromeLeCoz** [comment]
Thx, bug fixed. The 2-2 version is fully ok for me.
